### PR TITLE
Implemented tests for Content View registration via CLI

### LINKED ISSUE #681: Write tests  for Content View registration via CLI
Please write the tests for the following (already written) stubs:

* test_cv_subscribe_system

### PR DESCRIPTION
**Pull request by oshtaier**
Covered all variation mentioned in stub comments

Closes #681 

```
nosetests tests/foreman/cli/test_contentviews.py -m test_cv_subscribe_system
......
----------------------------------------------------------------------
Ran 6 tests in 686.940s

OK
```

### CODE REVIEW

File: tests/foreman/cli/test_contentviews.py [hunk 1]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+
+        @feature: Content Views
+
+        @assert: Content host can be subscribed to content view
+
+        """
+        new_org = make_org()
+        env = make_lifecycle_environment({u'organization-id': new_org['id']})
+        cv = make_content_view({u'organization-id': new_org['id']})
+        ContentView.publish({u'id': cv['id']})
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Better to assert that publish have succeeded.

File: tests/foreman/cli/test_contentviews.py [hunk 2]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+        result = ContentView.info({u'id': cv['id']})
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Better to assert that the info command succeeded.

File: tests/foreman/cli/test_contentviews.py [hunk 3]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+        cvv = result.stdout['versions'][0]
+        result = ContentView.version_promote({
+            u'id': cvv['id'],
+            u'to-lifecycle-environment-id': env['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+
+        result = ContentView.info({u'id': cv['id']})
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Better to assert that the info command succeeded.

File: tests/foreman/cli/test_contentviews.py [hunk 4]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+        self.assertEqual(result.stdout['content-host-count'], '0')
+
+        make_content_host({
+            u'name': gen_alphanumeric(),
+            u'organization-id': new_org['id'],
+            u'lifecycle-environment-id': env['id'],
+            u'content-view-id': cv['id'],
+        })
+        result = ContentView.info({u'id': cv['id']})
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Better to assert that the info command succeeded.

File: tests/foreman/cli/test_contentviews.py [hunk 5]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+            u'organization-id': TestContentView.rhel_content_org['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+        result = ContentView.info({u'id': cv['id']})
+        self.assertEqual(
+            result.stdout['yum-repositories'][0]['name'],
+            self.rhel_repo_name,
+        )
+
+        name = gen_string('alphanumeric')
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: If UTF8 is accepted then should use it as it have more complex character representation type.

File: tests/foreman/cli/test_contentviews.py [hunk 6]
```diff
@@ -1847,6 +1847,293 @@ def test_cv_publish_version_changes_in_source_env(self):
...
+            'content-view-id': cv['id'],
+            'type': 'rpm',
+            'inclusion': 'true',
+            'name': name,
+        })
+        self.assertEqual(result.return_code, 0)
+
+        result_rl = ContentView.filter_rule_create({
+            'content-view-id': cv['id'],
+            'name': gen_string('alphanumeric'),
```
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Would use UTF8 here too.

**Comment by elyezer** [comment]
Asserting for the return code is a good practice and will stop the test executing immediately if a command fail. I have commented on the first test method but others need to be updated.

Using UTF8 string type is not a blocker, but it is a better type to use because the multibyte character representation it has.

ACK, pending comments.

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
ACK

**Pull request merged**